content/api/node.md

```diff
@@ -1,7 +1,117 @@
 ---
 title: Node.js API
 ---
-> webpack()
-> run()
-> compile()
-> watch()
+Webpack provides a Node.js API which can be used directly in Node.js runtime. To get started with the Node.js API, require the `webpack` module:
+
+``` js
+const webpack = require("webpack");
+```
+
+Or if you’re using the ES6 modules interface:
+
+``` js
+import webpack from "webpack";
+```
+
+## Quick Run
+
+The imported `webpack` function is fed a Webpack [Configuration Object](/configuration/) and runs the webpack compiler if a callback function is provided:
+
+``` js-with-links
+const webpack = require("webpack");
+
+webpack({
+  // [Configuration Object](/configuration/)
+}, (err, [stats](#stats-object)) => {
+  // Done
+  console.log(stats);
+});
+```
+
+**Note** that you can provide the `webpack` function with an array of configurations:
+
+``` js-with-links
+webpack([
+  { /* Configuration Object */ },
+  { /* Configuration Object */ },
+  { /* Configuration Object */ }
+], (err, [stats](#stats-object)) => {
+  // ...
+});
+```
+
+## Compiler Instance
+
+If you don’t pass the `webpack` runner function a callback, it will return a Webpack `Compiler` intance. This instance can be used to manually trigger the Webpack runner or have it build and watch for changes. Much like the [CLI](/api/cli/) Api. The `Compiler` intance provides the following methods:
+
+* `.run(callback)`
+* `.watch(watchOptions, handler)`
+
+## Run
+
+Calling the `run` method on the `Compiler` intance is much like the quick run method mentioned above:
+
+``` js-with-links
+const webpack = require("webpack");
+
+const compiler = webpack({
+  // [Configuration Object](/configuration/)
+});
+
+compiler.run((err, [stats](#stats-object)) => {
+  // ...
+});
+```
+
+## Watch
+
+Calling the `watch` method, triggers the Webpack runner, but then watches for changes (much like CLI: `webpack --watch`), as soon as Webpack detects a change, runs again:
+
+``` js-with-links
+watch([watchOptions](#watch-options), callback)
+```
+
+``` js-with-links-with-details
+const webpack = require("webpack");
+
+const compiler = webpack({
+  // [Configuration Object](/configuration/)
+});
+
+const watcher = compiler.watch({
+  <details><summary>/* watchOptions */</summary>
+  aggregateTimeout: 300,
+  poll: undefined
+  </details>
+}, (err, [stats](#stats-object)) => {
+  // Print watch/build result here...
+  console.log(stats);
+});
+```
+
+### Watch Options
+
+First parameter is watchOptions object:
```

Comment:
This is also documented [here](https://webpack.js.org/configuration/watch/#watchoptions). Maybe only show a code example here, and link to the WatchOptions documentation for more information?